pages/utilities/text-difference.tsx

```diff
@@ -0,0 +1,108 @@
+import { useEffect, useState } from "react";
+import { Textarea } from "@/components/ds/TextareaComponent";
+import PageHeader from "@/components/PageHeader";
+import { Card } from "@/components/ds/CardComponent";
+import { Label } from "@/components/ds/LabelComponent";
+import Header from "@/components/Header";
+import { CMDK } from "@/components/CMDK";
+import CallToActionGrid from "@/components/CallToActionGrid";
+import Meta from "@/components/Meta";
+import GitHubContribution from "@/components/GitHubContribution";
+import { calculateDiff } from "@/components/utils/text-difference.utils";
+
+type DiffResult = {
+  text: string;
+  type: string;
+};
+
+export default function TextDifference() {
+  const [input1, setInput1] = useState("");
+  const [input2, setInput2] = useState("");
+  const [diffResults, setDiffResults] = useState<DiffResult[]>([]);
```

Comment:
Instead of state and an effect that updates it, please use `useMemo` — it's more idiomatic and created exactly for this purpose.